boards/FSM/Src/Io/Io_WheelSpeedSensors.c

```diff
@@ -0,0 +1,74 @@
+#include <assert.h>
+#include <math.h>
+
+#include "Io_WheelSpeedSensors.h"
+#include "Io_SharedFreqOnlyPwmInput.h"
+#include "main.h"
+
+// Note: Unit for length is measured in metres unless specified
+static const float  MPS_TO_KPH_CONVERSION_FACTOR = 3.6;
+static const size_t RELUCTOR_RING_TOOTH_COUNT    = 48;
+static const float  TIRE_DIAMETER                = 0.4572;
+static const float  ARC_LENGTH_PER_RELUCTOR_TOOTH =
+    (float)((M_PI * TIRE_DIAMETER) / RELUCTOR_RING_TOOTH_COUNT);
+
+static struct FreqOnlyPwmInput *left_wheel_speed_sensor,
+    *right_wheel_speed_sensor;
+
+void Io_WheelSpeedSensors_Init(
+    TIM_HandleTypeDef *htim_left_wheel_speed_sensor,
+    TIM_HandleTypeDef *htim_right_wheel_speed_sensor)
+{
+    assert(htim_left_wheel_speed_sensor != NULL);
+    assert(htim_right_wheel_speed_sensor != NULL);
+
+    left_wheel_speed_sensor = Io_SharedFreqOnlyPwmInput_Create(
+        htim_left_wheel_speed_sensor, TIMx_FREQUENCY / TIM16_PRESCALER,
+        TIM_CHANNEL_1, TIM16_AUTO_RELOAD_REG, HAL_TIM_ACTIVE_CHANNEL_1);
+
+    right_wheel_speed_sensor = Io_SharedFreqOnlyPwmInput_Create(
+        htim_right_wheel_speed_sensor, TIMx_FREQUENCY / TIM17_PRESCALER,
+        TIM_CHANNEL_1, TIM17_AUTO_RELOAD_REG, HAL_TIM_ACTIVE_CHANNEL_1);
+}
+
+float Io_WheelSpeedSensors_GetLeftSpeed(void)
+{
+    return MPS_TO_KPH_CONVERSION_FACTOR * ARC_LENGTH_PER_RELUCTOR_TOOTH *
```

Comment:
Is the unit Rpm? We should append the unit to the name: `_GetLeftSpeedRpm`

Reply:
Ok fixed

Reply:
App code should have the kph suffix to be consistent